packages/perseus-editor/src/util/modernize-widgets-utils.ts

```diff
@@ -0,0 +1,205 @@
+import type {
+    NumericInputWidget,
+    PerseusRenderer,
+    PerseusWidgetsMap,
+} from "@khanacademy/perseus";
+
+type WidgetRenameMap = {
+    [oldKey: string]: string;
+};
+
+// This utils in this file are used to modernize our Perseus JSON structure,
+// so that we can convert deprecated widgets to their modern equivalents when
+// content creators use the Editor Page to update content containing these widgets.
+// Currently, we're only converting input-number to numeric-input,
+// but we can add more conversions here in the future.
+// Modernize the json content of a PerseusRenderer object
+// by converting deprecated widgets to their modern equivalents
+export const convertDeprecatedWidgets = (
+    json: PerseusRenderer,
+): PerseusRenderer => {
+    // If there's no json, or no input-number widgets, then we can return the json as is
+    if (!json || !conversionRequired(json)) {
+        return json;
+    }
+
+    // Currently we're only converting input-number to numeric-input,
+    // But we can add more conversions here in the future
+    const modernJson = inputNumberToNumericInput(json);
+
+    return modernJson;
+};
+
+const conversionRequired = (json: PerseusRenderer): boolean => {
+    const inputNumberRegex = /input-number \d+/; // We can add more regexes here in the future
+
+    // Check the content string for any top-level input-number widgets
+    if (inputNumberRegex.test(json.content)) {
+        return true;
+    }
```

Comment:
This might be over-engineering, but to aid in possible future conversions, what do you think about setting things up this way:
```
const widgetRegExes = [/input-number \d+/]; // Add your widget expression here

if (widgetRegExes.some(regex => regex.test(json.content))) {
    return true;
}
```